### ISSUE TITLE
iOS-App stuck in zoom mode after rotating screen from portrait to landscape

### ISSUE THREAD

**Issue by orgebar** [home-assistant/iOS#2856]
**iOS device model, version and app version**

Model Name: iPhone 12 Pro
Software Version: iOS 17.5.1 (21F90)
App version: 20240710.0

**Home Assistant Core Version**

2024.7.2

**Describe the bug**

Very often, if you rotate the screen from portrait to landscape mode (automatic rotation turned on in iOS control center), some kind of zoom is applied, which is not revertible, if the zoom mode is turned of in the companion apps settings. This happens very often randomly throughout the whole app (= isnt connected to specific views/pages like Dashboard views, Add on view, Settings view or else). Switching back to Portrait or refreshing the page doesnt help. You need to close the app completely and restart it to get the full screen again.

**To Reproduce**

Switch from Portrait to landscape view back and forth while using the app.

**Expected behavior**
Should not zoom at all, if zoom is disabled or at least it should be possible to zoom out to full screen once, even if zoom is disabled

**Screenshots**

![image](https://github.com/user-attachments/assets/ee0a23ea-23f1-46f9-b3d5-9fcc17fa89a2)

**Additional context**

This behaviour isn't new and I don't think its connected to the beta version, as I've observed this since i started (at least 1.5 years ago) in the different app versions. I updated the app regularly every time there was an update.

It must be a general iOS app problem, as the same behaviour can be observed using my iPads or other iPhone models.

**Comment by bgoncal** [comment]
Can you check if the same happens on iOS safari?

**Comment by bgoncal** [comment]
I'll be closing this issue for now, let me know if it only happens in iOS App, then I will reopen, otherwise please open an issue on frontend repo.

**Comment by orgebar** [comment]
Hi,

I tried it with Safari (on iPhone) for a while now but couldn‘t reproduce the error. Im quite convinced this behaviour is limited to the iOS HA App only, so please reopen the ticket :).

Thanks!

**Comment by Stooovie** [comment]
This happen 100% of the time, but only after a keyboard has been open. As in: the zoom-in bug doesn't appear until i open a keyboard (by tapping a text entry field). Then, any rotation zooms the UI in and doesn't return. HA 2025.6b5 on iOS 18.5 as of now, but it's a long-standing (years!) thing that sort of comes and goes. It's NOT related to a specific theme nor a 3rd-party keyboard.

[Here's a video](https://youtube.com/shorts/gI3TRe7G67s?si=ptMczSYadiAm2EF8)

**Comment by orgebar** [comment]
I'd be very happy, if this issue could be resolved :). And I really think its time to do this now! 😝

For me (and I cant be the only one) It's for sure the most annoying bug using the app, since you have to restart the app completely every time. As mentioned above, this bug exists for years now... Given 1 restart a day and 3 years usage I needed to restart the app about 3.300 times 😂

To Whoever is responsible for the App and has the ability to fix this: I'm begging you to do so - If you think about the necessary swipes and clicks, to restart and go back where you started, we all have already lost a wild number of our "total swipes/clicks before the doom scrolling thumb arthrosis will kick in with certainty"-balance. Please have mercy with our thumbs!!! 😂😝

**Comment by bgoncal** [comment]
Have you tried disabling "pinch to zoom" under companion App settings > General ?

**Comment by Stooovie** [comment]
Yes. The only effect is that the issue cannot be manually reverted and the app needs to be killed and restarted. The issue itself remains.

**Comment by bgoncal** [comment]
Ok, first for the @orgebar and @Stooovie can you share your current Core and App versions?

I am not being able to reproduce following your video @Stooovie but I'll keep trying with other devices I have

**Comment by orgebar** [comment]
I can confirm what @Stooovie reported.

Those are the versions running right now.

![image](https://github.com/user-attachments/assets/ee338a76-716f-4c5a-bc14-c7ec8c35f6ee)

As mentioned before. The error occured with any combination of core an frontend version over the past 2-3 years (and different end devices iPhone 12Pro&16Pro, iPad 2020). You should be able to reproduce it in any given setup using the iOS app. 

1. Use the Keyboard at least once
2. flip the device so it auto rotats to landscape mode
3. Flip it back

It Happens not every time. Try it on different sites. At some point it will for sure (randomly) freeze with zoom

**Comment by bgoncal** [comment]
I'll be trying with more devices soon, but thats from my iPhone 16 Pro, can't seem to reproduce.
https://github.com/user-attachments/assets/9d59b856-42d2-4d1f-b82c-e8a7dbe8eb27

**Comment by Stooovie** [comment]
> Ok, first for the [@orgebar](https://github.com/orgebar) and [@Stooovie](https://github.com/Stooovie) can you share your current Core and App versions?
> 
> I am not being able to reproduce following your video [@Stooovie](https://github.com/Stooovie) but I'll keep trying with other devices I have

App 2025.5, Core 2025.6.1, but both are largely irrelevant, the issue goes back years and years. It is not dependent on a theme or 3rd-party keyboard. It happens every time I open a keyboard within the HA app and then rotate a device. It does not matter if the keyboard is open at the time of rotation. If the keyboard has not been open since launching the HA app, issue does not happen.

**Comment by bgoncal** [comment]
https://openradar.appspot.com/44188896
https://github.com/react-native-webview/react-native-webview/issues/3477

I see some reports similar to that, I'll investigate if I can provide a workaround, although it looks like something on Apple's plate

**Comment by Stooovie** [comment]
I can me more specific - I have now discovered the issue (probably) only happens with Page Zoom set to my favorite 85%. If I set it to 100% (default), issue does not happen.

**Comment by bgoncal** [comment]
@orgebar can you confirm the above as well?

**Comment by Stooovie** [comment]
> [@orgebar](https://github.com/orgebar) can you confirm the above as well?

In case you missed my last comment - try it with Page Zoom in the app set to 85% (which I prefer). It doesn't seem to happen with Page Zoom set to 100%. This might be the condition we missed.

I can obviously set it to 100% in the meantime but I'd really like to have this fixed :)

**Comment by bgoncal** [comment]
I'll let you know if I find a workaround, but this feature (page zoom) was being considered for removal btw, since Apps are not expected to zoom like browsers, for now they are there as helpers but are not expected to stay for long term.

**Comment by Stooovie** [comment]
That's too bad, so how am I supposed to fit more of the dashboard onto the display? The default 100% zoom is pretty huge on an iPhone 13 Pro and the app doesn't respond to iOS' built-in text size options.

85 vs 100 %
![Image](https://github.com/user-attachments/assets/b8e16c93-cdb7-48a3-bec2-f7b131f895e0)
![Image](https://github.com/user-attachments/assets/ed4ecd1c-60bd-41dc-9aac-ca2a8789b38f)

**Comment by Stooovie** [comment]
Weirdly enough, 75% Zoom does not exhibit the issue. Only 85% does.

**Comment by orgebar** [comment]
> @orgebar can you confirm the above as well?

I can confirm that this happens when using zoom 85% bc I use this Setting on all devices all the time.

I can also confirm that it doesnt happen anymore when using 100%.

**Comment by Stooovie** [comment]
I'm happy we fould the missing piece

**Comment by orgebar** [comment]
> That's too bad, so how am I supposed to fit more of the dashboard onto the display? The default 100% zoom is pretty huge on an iPhone 13 Pro and the app doesn't respond to iOS' built-in text size options.
> 
> 85 vs 100 %
> ![Image](https://github.com/user-attachments/assets/b8e16c93-cdb7-48a3-bec2-f7b131f895e0)
> ![Image](https://github.com/user-attachments/assets/ed4ecd1c-60bd-41dc-9aac-ca2a8789b38f)

I strongly Support that opinion: 100% is way too big on any device and losing the Option to zoom to smaller would be a major drawback in my opinion if not to say a big mistake. Its so big, It looks like everything on my grandfathers phone, where I set the font to maximum so that he can see anything, but the result is that you dont see anything except the font. Please don't do this.

**Comment by orgebar** [comment]
> I'm happy we fould the missing piece

Congrats to you for finding the reason for the issue. That was a tough one!

And who knows... maybe this will be the reason for not losing zoom too.

**Comment by bgoncal** [comment]
I understand your opinion but "100%" is how Home Assistant is being designed and any other zoom size is entirely a bonus and not the intended UI, if you don't agree with it's sizing you need to open a feature request explaining your needs and the reasoning behind. 

For example, looking at my Home Screen, it looks like a normal smart home App screen, 2 columns with each tile of the height of a finger touch, you can compare to Apple Home for example.

![Image](https://github.com/user-attachments/assets/8d3d576d-0bbb-43cf-8a66-6184729f40ac)

![Image](https://github.com/user-attachments/assets/4f535118-bbb6-4135-9fe0-13cb62b827cf)

**Comment by bgoncal** [comment]
Good news for you, I discussed with one of the designers and zoom won't be removed.

**Comment by Stooovie** [comment]
That's great to hear :) I'd love to know why the bug only manifests at 85% zoom and not lower or higher.

**Comment by bgoncal** [comment]
Since iOS 26 I cannot reproduce the issue anymore, can you?

**Comment by bgoncal** [comment]
And if it is still reproducible, can you once again try to reproduce on iOS Safari as well?

**Comment by z3r0bytes** [comment]
Hi 

I can confirm that in 75 & 85% zoom on iOS 26 it is still happening. Only after you have interacted with the keyboard. 

Just tested it now. 
Installation method
Home Assistant OS
Core
2026.2.3
Supervisor
2026.02.3
Operating System
17.1
Frontend
20260128.6
Companion app
2026.2.1 (2026.1766)
